### ISSUE TITLE
Allow Jetty's thread pool to be configured via the environment

### ISSUE THREAD

**Issue by dmwb** [spring-projects/spring-boot#5314]
Hello,

I'm wondering, if it is possible to have some basic configuration for embedded Jetty in application.properties. Right now to customize thread pool and idle timeout, we're creating bean, which is not convenient...

```
    @Bean
    public JettyEmbeddedServletContainerFactory jettyEmbeddedServletContainerFactory() {
        final JettyEmbeddedServletContainerFactory factory = new JettyEmbeddedServletContainerFactory(serverProperties.getPort());
        factory.addServerCustomizers((Server server) -> {
            final QueuedThreadPool threadPool = server.getBean(QueuedThreadPool.class);
            threadPool.setMinThreads(appProperties.getMinThreads());
            threadPool.setMaxThreads(appProperties.getMaxThreads());
            threadPool.setIdleTimeout(appProperties.getIdleTimeout());
        });
        return factory;
    }
```

Could we have something similar like for tomcat and undertow. F.e.

```
server.jetty.max-threads
server.jetty.min-threads
server.jetty.idle-timeout
```

Or make it more generic. I'm pretty sure each servlet container has ability to configure such properties. 

Thanks!

**Comment by wilkinsona** [comment]
See also #5324

**Comment by wilkinsona** [comment]
We already have the following properties:

```
server.tomcat.max-threads
server.undertow.io-threads
server.undertow.worker-threads
```

Tomcat's `AbstractProtocol` also supports configuring the minimum number of spare threads for which we do not have a property.

Jetty uses a `QueueThreadPool` by default. It has three simple properties:

```
maxThreads (default 200)
minThreads (default 8)
idleTimeout (default 60000)
```

It also allows its `BlockQueue` to be configured. It defaults to a `BlockingArrayQueue` which is unbounded with an initial capacity of the bigger of 8 and the minimum number of threads.

**Comment by dmwb** [comment]
Thanks in advance! Jetty is widely used and having these common parameters configured via application properties etc. will make Boot more efficient...

**Comment by btridip-in** [comment]
Is there a way to specify the max threads via the configuration (.yml) file to control how many concurrent requests can be serviced by the instance at any point in time?

**Comment by wilkinsona** [comment]
@btridip-in No, that's what this issue is about and it's still open. We have made it easier to configure it programmatically. See #5324 that was included in 1.4.

**Comment by mbhave** [comment]
Closing in favor of PR #17871.
